### PR TITLE
feat: optimize search store by removing inline EXISTS

### PR DESCRIPTION
**Pull request by sjaanus**
Instead of running exists on every row, we are joining the exists, which runs the query only once.
This decreased load time on my huge dataset from 2000ms to 200ms.

Also added tests that values still come through as expected.

### CODE REVIEW

File: src/lib/features/feature-search/feature-search-store.ts [hunk 1]
```diff
@@ -161,12 +164,6 @@ class FeatureSearchStore implements IFeatureSearchStore {
 
                 selectColumns = [
                     ...selectColumns,
-                    this.db.raw(
```
Comment (sjaanus) [inline src/lib/features/feature-search/feature-search-store.ts:164]: This is the part we removing

File: src/lib/features/feature-search/feature-search-store.ts [hunk 2]
```diff
@@ -351,6 +338,72 @@ class FeatureSearchStore implements IFeatureSearchStore {
...
+        queryBuilder.leftJoin('metrics', (qb) => {
+            qb.on(
+                'metric_environment',
+                '=',
+                'ranked_features.environment',
+            ).andOn('metric_feature_name', '=', 'ranked_features.feature_name');
+        });
+    }
+
+    private queryStrategiesByEnvironment(queryBuilder: Knex.QueryBuilder) {
```
Comment (sjaanus) [inline src/lib/features/feature-search/feature-search-store.ts:356]: This is what I am adding

File: src/lib/features/feature-search/feature-search-store.ts [hunk 3]
```diff
@@ -351,6 +338,72 @@ class FeatureSearchStore implements IFeatureSearchStore {
         };
     }
 
+    private queryExtraData(queryBuilder: Knex.QueryBuilder) {
+        this.queryMetrics(queryBuilder);
+        this.queryStrategiesByEnvironment(queryBuilder);
+    }
+
+    private queryMetrics(queryBuilder: Knex.QueryBuilder) {
```
Comment (sjaanus) [inline src/lib/features/feature-search/feature-search-store.ts:346]: Also very small refactor, moving this also out of main join.

Comment (sjaanus) [reply src/lib/features/feature-search/feature-search-store.ts:346]: Reducing the big method by 20 lines.

**Review by kwasniew — APPROVED** [review]

**Pull request merged**